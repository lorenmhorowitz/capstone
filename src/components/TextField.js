import "../css/Component.css";
import MuiTextField from "@material-ui/core/TextField";
import React, { Component } from "react";

class TextField extends Component {
  render() {
    const { label } = this.props;
    return (
      <div>
        <form id="textField">
          <MuiTextField id="fieldStyle" label={label} className="form_label" />
        </form>
      </div>
    );
  }
}

export default TextField;
