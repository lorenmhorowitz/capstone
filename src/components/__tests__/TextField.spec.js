import React from "react";
import TextField from "../TextField";
import { shallow } from "enzyme";

describe("TextField component renders properly", () => {
  it("renders with a label passed in", () => {
    let wrapper = shallow(<TextField label="email" />);
    expect(wrapper).toMatchSnapshot();
  });
});
