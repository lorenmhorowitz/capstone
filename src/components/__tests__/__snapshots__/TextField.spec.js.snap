// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField component renders properly renders with a label passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextField
    label="email"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        id="textField"
      >
        <TextField
          id="fieldStyle"
          label="email"
          name="email"
          required={false}
          select={false}
          type="email"
          variant="standard"
        />
      </form>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <TextField
          id="fieldStyle"
          label="email"
          name="email"
          required={false}
          select={false}
          type="email"
          variant="standard"
        />,
        "id": "textField",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "id": "fieldStyle",
          "label": "email",
          "name": "email",
          "required": false,
          "select": false,
          "type": "email",
          "variant": "standard",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          id="textField"
        >
          <TextField
            id="fieldStyle"
            label="email"
            name="email"
            required={false}
            select={false}
            type="email"
            variant="standard"
          />
        </form>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <TextField
            id="fieldStyle"
            label="email"
            name="email"
            required={false}
            select={false}
            type="email"
            variant="standard"
          />,
          "id": "textField",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "id": "fieldStyle",
            "label": "email",
            "name": "email",
            "required": false,
            "select": false,
            "type": "email",
            "variant": "standard",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
